beacon-chain/db/filesystem/blob.go

```diff
@@ -279,20 +279,30 @@ func (bs *BlobStorage) Prune(pruneBefore primitives.Slot) error {
 // processFolder will delete the folder of blobs if the blob slot is outside the
 // retention period. We determine the slot by looking at the first blob in the folder.
 func (bs *BlobStorage) processFolder(folder string, pruneBefore primitives.Slot) (int, error) {
-	f, err := bs.fs.Open(filepath.Join(folder, "0."+sszExt))
-	if err != nil {
-		return 0, err
-	}
-	defer func() {
-		if err := f.Close(); err != nil {
-			log.WithError(err).Errorf("Could not close blob file")
+	var f afero.File
+	var err error
+	var slot primitives.Slot
+
+	for i := 0; i < fieldparams.MaxBlobsPerBlock; i++ {
+		f, err = bs.fs.Open(filepath.Join(folder, fmt.Sprintf("%d.%s", i, sszExt)))
+		if err == nil {
+			slot, err = slotFromBlob(f)
+			if err != nil {
+				return 0, err
+			}
+			break
 		}
-	}()
+		if f != nil {
+			if err := f.Close(); err != nil {
```

Comment:
Looks like you will only close f in some cases. If you hit the break on 293, then the file isn't closed.

Reply:
Ah. Thanks!